### DISCUSSION TITLE
CommunityMVVM Source Generator - Go To Definition F12 in Visual Studio

### DISCUSSION THREAD

**Discussion by gregvis** [dotnet/roslyn#65463] [Ideas]
I posted this in Community Toolkit discussions, but it was suggested I post here. 

I use F12 quite a bit to jump around my code in Visual Studio, but with MVVM generators, it doesn't go where I want it to go

Example

```
[ObservableProperty]
string myProperty


MyProperty = "example"
```

if i go to definition (F12) for MyProperty, it takes me to the generated g.cs file instead of to my code (which should go to "myProperty" . Is there a way to fix this in Visual Studio?  

Same example for in xaml

If I have the command

```
[RelayCommand]
void DoStuff() {}
```

and in xaml I reference it
```
<Button Command={Binding DoStuffCommand} />
```
When I click F12 on DoStuffCommand, I want it to go to DoStuff instead of the g.cs file

Is this possible?

**Comment by weltkante** [comment]
> if i go to definition (F12) for MyProperty, it takes me to the generated g.cs file instead of to my code

I found that often you can do ctrl+F12 to go to the other half of the declaration.
